pkg/util/pki/csr_test.go

```diff
@@ -669,15 +670,91 @@ func TestSignCSRTemplate(t *testing.T) {
 				// into strings and do a textual diff.
 				expected, _ := DecodeX509CertificateBytes(test.expectedCertPem)
 				actual, _ := DecodeX509CertificateBytes(actualBundle.ChainPEM)
+
 				assert.Equal(t, expected.Subject.String(), actual.Subject.String())
 			}
+
 			if !bytes.Equal(test.expectedCaCertPem, actualBundle.CAPEM) {
 				// To help us identify where the mismatch is, we decode turn the
 				// into strings and do a textual diff.
 				expected, _ := DecodeX509CertificateBytes(test.expectedCaCertPem)
 				actual, _ := DecodeX509CertificateBytes(actualBundle.CAPEM)
+
 				assert.Equal(t, expected.Subject.String(), actual.Subject.String())
 			}
 		})
 	}
 }
+
+func TestEncodeX509Chain(t *testing.T) {
+	root := mustCreateBundle(t, nil, "root")
+	intA1 := mustCreateBundle(t, root, "intA-1")
+	intA2 := mustCreateBundle(t, intA1, "intA-2")
+	leafA1 := mustCreateBundle(t, intA1, "leaf-a1")
+	leafA2 := mustCreateBundle(t, intA2, "leaf-a2")
+	leafInterCN := mustCreateBundle(t, intA1, intA1.cert.Subject.CommonName)
```

Comment:
suggestion: I struggled to picture the tree, would it be worth adding a diagram? 

```go
	//  	                        root                  iss = issuer               
	//                          iss: CN=root              sub = subject              
	//                          sub: CN=root                            
	//                                |                                 
	//                             intA-1                               
	//                          iss: CN=root                            
	//                          sub: CN=intA-1                          
	//                                |                                 
	//        +-----------------------+-------------------------+       
	//        |                       |                         |       
	//     leaf-a1                 intA-2                  leafInterCN  
	//  iss: CN=intA-1          iss: CN=intA-1            iss: CN=intA-1
	//  sub: leaf-a1            sub: CN=intA-2            sub: CN=intA-1
	//                                |                                 
	//                             leaf-a2                              
	//                          iss: CN=intA-2                          
	//                          sub: CN=leaf-a2  
	//
```

Does it match what you intended?

Reply:
Argh, it took me over 1 hour to review and I now realized that my review is moot since this is now merged 😅